File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/database/mapper/RealmLocationMapper.java
package com.lattis.ellipse.data.database.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.database.base.AbstractRealmDataMapper;
import com.lattis.ellipse.data.database.model.RealmLocation;
import com.lattis.ellipse.domain.model.Location;

import javax.inject.Inject;

public class RealmLocationMapper extends AbstractRealmDataMapper<Location,RealmLocation> {

    @Inject
    public RealmLocationMapper() {}

    @NonNull
    @Override
    public RealmLocation mapIn(@NonNull Location location) {
        RealmLocation realmLocation = new RealmLocation();
        location.setLatitude(location.getLatitude());
        location.setLongitude(location.getLatitude());
        return realmLocation;
    }

    @NonNull
    @Override
    public Location mapOut(@NonNull RealmLocation realmLocation) {
        Location location = new Location();
        location.setLatitude(realmLocation.getLatitude());
        location.setLongitude(realmLocation.getLongitude());
        return location;
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/database/mapper/RealmLocationMapper.kt
package com.lattis.data.database.mapper

import com.lattis.data.database.base.AbstractRealmDataMapper
import com.lattis.data.database.model.RealmLocation
import com.lattis.domain.models.Location
import javax.inject.Inject

class RealmLocationMapper @Inject constructor() :
    AbstractRealmDataMapper<Location, RealmLocation>() {
    override fun mapIn(location: Location): RealmLocation {
        val realmLocation = RealmLocation()
        location.latitude = location.latitude
        location.longitude = location.latitude
        return realmLocation
    }

    override fun mapOut(realmLocation: RealmLocation): Location {
        val location =
            Location()
        location.latitude = realmLocation.latitude
        location.longitude = realmLocation.longitude
        return location
    }
}